src/resolve/resolved-dns-query.c

```diff
@@ -517,7 +517,7 @@ static int manager_validate_and_mangle_question(Manager *manager, DnsQuestion **
 
         if (!*question) {
                 *ret_allocated = NULL;
-                return 0;
+                return -ENOANO;
```

Comment:
Why? Please explain in the commit message.
If this is the way to go, then please do not initialize `ret_allocated`.

Reply:
Yeah, this was an old commit. Was reverted.